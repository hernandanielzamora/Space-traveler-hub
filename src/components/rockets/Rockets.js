import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import RocketList from './RocketList';
import { fetchingRocketsInfo } from '../../redux/rocket/RocketSlice';
import classes from './Rockets.module.css';

let saveReserve = false;
const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);

  const dispatch = useDispatch();

  useEffect(() => {
    if (saveReserve === false) {
      saveReserve = true;
      dispatch(fetchingRocketsInfo());
    }
  }, [dispatch]);

  return (
    <main className={classes.rockets}>
      <RocketList rockets={rockets} />
    </main>
  );
};

export default Rockets;
