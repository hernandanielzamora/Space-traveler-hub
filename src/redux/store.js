import { configureStore } from '@reduxjs/toolkit';

import rocketReducer from './rocket/RocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
  },
});

export default store;
