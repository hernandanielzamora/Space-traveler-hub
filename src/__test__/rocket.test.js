import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../components/rockets/Rockets';
import store from '../redux/store';

it('Test RocketContainer renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Rockets />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
