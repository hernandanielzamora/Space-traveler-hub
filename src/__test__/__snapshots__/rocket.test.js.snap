// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RocketContainer renders correctly 1`] = `
<main
  className="rockets"
>
  <ul />
</main>
`;
